### ISSUE TITLE
Opening '-' for append should return stdout, not stdin

### ISSUE THREAD

**Issue by weeble** [pallets/click#562]
Currently, passing mode='a' to a File type argument is allowed, but will result in "io.UnsupportedOperation: not writable" if the user specifies '-' and we ever try to write to the file. It would make more sense to return sys.stdout than sys.stdin in this situation.

Example:

```
$ cat demo.py
import click

@click.command()
@click.argument('appendfile', type=click.File(mode='a'))
def cli(appendfile):
    print('Hello', file=appendfile)

if __name__ == '__main__':
    cli()
$ cat foobar
Hello
Hello
$ python3 demo.py foobar
$ cat foobar
Hello
Hello
Hello
$ python3 demo.py -
Traceback (most recent call last):
  File "demo.py", line 9, in <module>
    cli()
  File "/usr/local/lib/python3.5/site-packages/click/core.py", line 700, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/click/core.py", line 680, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.5/site-packages/click/core.py", line 873, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.5/site-packages/click/core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "demo.py", line 6, in cli
    print('Hello', file=appendfile)
io.UnsupportedOperation: not writable
```

**Comment by blueyed** [comment]
Fixed in b8f5744820a1359291f11ae8bf8ad85313da000c.

**Comment by fdavis** [comment]
Closing because it was fixed in https://github.com/pallets/click/commit/b8f5744820a1359291f11ae8bf8ad85313da000c
